module/functions/Invoke-CommandWithRetry.ps1

```diff
@@ -0,0 +1,66 @@
+# <copyright file="Invoke-CommandWithRetry.ps1" company="Endjin Limited">
+# Copyright (c) Endjin Limited. All rights reserved.
+# </copyright>
+
+<#
+.SYNOPSIS
+Provides retry logic for PowerShell ScriptBlock execution.
+
+.DESCRIPTION
+Provides retry logic for PowerShell ScriptBlock execution.
+
+.PARAMETER Command
+Sets the scriptblock to be executed.
+
+.PARAMETER RetryCount
+Sets the maximum retry attempts.
+
+.PARAMETER RetryDelay
+Sets the delay (in seconds) between retry attempts.
+
+#>
+function Invoke-CommandWithRetry
```

Comment:
We should think about creating a core PS package with these types of functions that we can re-use in other places. We actually have a [blog post](https://endjin.com/blog/2014/07/how-to-retry-commands-in-powershell) on our site from 2014 describing how to write similar PowerShell retry functionality!

Reply:
Agreed, we can do a pass a later date for anything that is even more generic.

I've updated this version with some aspects from the one you linked to, but am leaving the bulk of it as we need to get the result from the retryable operation.